transform/mattermost-analytics/models/intermediate/hightouch/int_stripe_onprem_subscriptions.sql

```diff
@@ -0,0 +1,33 @@
+{{config({
+    "materialized": 'table',
+    "unique_key":"invoice_id",
+  })
+}}
+
+WITH payment AS (
+    SELECT
+        invoices.charge,
+        invoices.subscription,
+        'ONL' || invoices.number AS invoice_number,
+        invoices.number as stripe_invoice_number,
+        invoices.id as invoice_id,
+        invoices.created_at as invoice_created_at,
+        ROW_NUMBER() OVER (PARTITION BY invoices.subscription ORDER BY invoices.created_at) as row_num
+    FROM {{ ref('stg_stripe__invoices') }} invoices
+), subscriptions AS (
+    SELECT
+        p.sku as sku,
+        payment.charge as stripe_charge_id,
+        payment.invoice_number,
+        payment.stripe_invoice_number,
+        ili.quantity as invoice_quantity,
+        ili.amount as invoice_amount,
+        ROW_NUMBER() OVER (PARTITION BY s.subscription_id, payment.stripe_invoice_number order by payment.invoice_created_at) as row_num
+        , s.*
+    FROM {{ ref('stg_stripe__subscriptions') }} s
+    JOIN {{ ref('stg_stripe__products') }} p ON s.product = p.product_id
+    JOIN payment ON s.subscription_id = payment.subscription 
+    JOIN {{ ref('stg_stripe__invoice_line_items') }} ili ON payment.invoice_id = ili.invoice
+    WHERE p.name not like '%Cloud%' and amount > 0
```

Comment:
Perhaps use `ILIKE '%Cloud%'` to match ignoring case.

It would also be nice to add a comment on why `amount > 0`.

Reply:
The amount > 0 is necessary, when the customer pays for an expansion they first get refunded the original purchase amount and get charged for the new amount based on the new number of seats purchased. So in any subscription where an expansion has happened, we will see 3 rows of data in `invoice_line_items`.

1) The original purchase amount
2) The refunded amount which is < 0 (this is not required in SFDC, after I spoke to John)
3) The new amount charged.

We are only required to update in SFDC the amounts > 0.

Reply:
Can we capture this as a comment either in SQL or in the model description? It looks like useful information that's not obvious to guess.

Reply:
Added this to the model description